framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -435,22 +443,24 @@ private AnnotationMirror createBooleanAnnotationMirror(List<Boolean> values) {
 
     private List<Number> calculateNumericalUnaryOp(
             Node operand, NumericalUnaryOps op, TransferInput<CFValue, CFStore> p) {
-        List<? extends Number> lefts = getNumericalValues(operand, p);
         List<Number> resultValues = new ArrayList<>();
-        for (Number left : lefts) {
-            NumberMath<?> nmLeft = NumberMath.getNumberMath(left);
-            switch (op) {
-                case PLUS:
-                    resultValues.add(nmLeft.unaryPlus());
-                    break;
-                case MINUS:
-                    resultValues.add(nmLeft.unaryMinus());
-                    break;
-                case BITWISE_COMPLEMENT:
-                    resultValues.add(nmLeft.bitwiseComplement());
-                    break;
-                default:
-                    throw new UnsupportedOperationException();
+        if (!isIntRange(operand, p)) {
```

Comment:
Just to be clear, the control flow type refinement is NOT implemented in this PR (we have the calculation implemented in Range.java, but not used in the checker). In the checker, the result of a comparison operation is a `RegularTransferResult`, i.e. a `@BoolVal`, but not a `ConditionalTransferResult`, i.e. control flow type refinement. Please refer to https://github.com/typetools/checker-framework/pull/994/files#diff-0cd2d563914222b6c6044ca08956e7fcR511

I've implemented the control flow type refinement here as an experiment: https://github.com/JasonMrX/IntRange-Checker/blob/master/src/intrange/IntRangeTransfer.java#L285, which looks good. 

Just want to make sure if we want this in the checker.